### PR TITLE
Unifier - add support for email list

### PR DESCRIPTION
**Pull request by daryakoval**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/26923

## Description
When provided multiple support email, each email should be written in it's own bullet.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by daryakoval** [comment]
@roysagi please review only CHANGELOG.

File: demisto_sdk/commands/unify/unifier.py [hunk 1]
```diff
@@ -443,7 +443,11 @@ def add_contributors_support(self, unified_yml: Dict, contributor_type: str, con
         else:
             contributor_description = CONTRIBUTOR_DETAILED_DESC.format(contributor_type.capitalize(), author)
             if contributor_email:
-                contributor_description += f'\n- **Email**: [{contributor_email}](mailto:{contributor_email})'
+                if isinstance(contributor_email, list):
+                    for email in contributor_email:
+                        contributor_description += f'\n- **Email**: [{email}](mailto:{email})'
+                else:
+                    contributor_description += f'\n- **Email**: [{contributor_email}](mailto:{contributor_email})'
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/unify/unifier.py:450]:
```suggestion
                contributor_email = contributor_email if isinstance(contributor_email, list) else [contributor_email]:
                    for email in contributor_email:
                        contributor_description += f'\n- **Email**: [{email}](mailto:{email})'
```

File: demisto_sdk/commands/unify/tests/unifier_test.py [hunk 2]
```diff
@@ -789,6 +806,30 @@ def test_unify_partner_contributed_pack_no_email(mocker, repo):
...
 
 
+def test_unify_contributor_emails_list(mocker, repo):
+    """
+    Given
+        - Partner contributed pack with email list and url in the support details.
+    When
+        - Running unify on it.
+    Then
+        - Ensure unify create unified file with partner support notes, email list show each email in it's own bullet.
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/unify/tests/unifier_test.py:816]:
```suggestion
        - Ensure unify create a unified file with partner support email list.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Changelog
 * Added validation for approved content pack usecases and tags.
 * Added new code validations for *CommonServerPython* import to `XSOAR-linter`.
+* For partner supported content packs, improved the support email formatting in the integration detailed description.
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:4]:
```suggestion
* For partner supported content packs, added support for a list of emails.
```

File: demisto_sdk/commands/unify/unifier.py [hunk 4]
```diff
@@ -443,7 +444,9 @@ def add_contributors_support(self, unified_yml: Dict, contributor_type: str, con
         else:
             contributor_description = CONTRIBUTOR_DETAILED_DESC.format(contributor_type.capitalize(), author)
             if contributor_email:
-                contributor_description += f'\n- **Email**: [{contributor_email}](mailto:{contributor_email})'
+                email_list: List[str] = contributor_email if isinstance(contributor_email, list) else [contributor_email]
+                for email in email_list:
+                    contributor_description += f'\n- **Email**: [{email}](mailto:{email})'
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/unify/unifier.py:449]: This seems wrong. `contributor_email` is a string even if the metadata contains a list of emails seems like we need to split here...

Comment (daryakoval) [reply demisto_sdk/commands/unify/unifier.py:449]: spoke with you on slack. 
we receive emails from json file, therefore we will receive a list of strings or one string.

Comment (ShahafBenYakir) [reply demisto_sdk/commands/unify/unifier.py:449]: Not necessarily true, as I said in slack we need to support both (lists or lists of strings) something like argToList - I think we already have a function in SDK tools for it. FYI in the metadata given in the issue example, the emails were a string

Comment (roysagi) [reply demisto_sdk/commands/unify/unifier.py:449]: @ShahafBenYakir - not sure we have that in SDK tools but could be a good addition.

Comment (daryakoval) [reply demisto_sdk/commands/unify/unifier.py:449]: added function

File: demisto_sdk/commands/common/tests/tools_test.py [hunk 5]
```diff
@@ -601,3 +602,23 @@ def test_get_ignore_pack_tests__ignore_missing_test(tmpdir, mocker):
...
+def test_arg_to_array(arg: Union[List[str], str], expected_result: List[str]):
+    """
+        Given
+        - arg - string or list of strings. For example: "a,b" -> ['a','b']
+        or "[a,b]" -> ['a','b']
+        When
+        - Convert given string to list of strings, for example at unify, add_contributors_support, when adding email.
+        Then:
+        - returns a list with the skipped tests
+        """
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/common/tests/tools_test.py:622]:
```suggestion
        Given
        - String or list of strings.
        Case a: comma-separated string.
        Case b: a string representing a list.
        Case c: python list.
        Case d: empty string.
        Case e: empty list.

        When
        - Convert given string to list of strings, for example at unify.add_contributors_support.
        
        Then:
        - Ensure a Python list is returned with the relevant values.
        """
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -3,7 +3,8 @@
 * Added new code validations for *CommonServerPython* import to `XSOAR-linter`.
 * Added *default value* and *predefined values* to argument description in **generate-docs** command.
 * Added a new validation that checks if *get-mapping-fields* command exists if the intgeration schema has *{ismappable: true}* in **validate** command.
-
+* For partner supported content packs, added support for a list of emails.
+* Added arg_to_array function to sdk-tools.
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:7]: No need for the last note, it is not relevant for the SDK users.

File: demisto_sdk/commands/unify/unifier.py [hunk 7]
```diff
@@ -443,7 +444,9 @@ def add_contributors_support(self, unified_yml: Dict, contributor_type: str, con
         else:
             contributor_description = CONTRIBUTOR_DETAILED_DESC.format(contributor_type.capitalize(), author)
             if contributor_email:
-                contributor_description += f'\n- **Email**: [{contributor_email}](mailto:{contributor_email})'
+                email_list: List[str] = contributor_email if isinstance(contributor_email, list) else [contributor_email]
```
Comment (ShahafBenYakir) [inline demisto_sdk/commands/unify/unifier.py:447]: Where is the new function?

File: demisto_sdk/commands/common/tools.py [hunk 8]
```diff
@@ -1430,3 +1430,24 @@ def get_demisto_version(demisto_client: demisto_client) -> str:
     resp = demisto_client.generic_request('/about', 'GET')
     about_data = json.loads(resp[0].replace("'", '"'))
     return parse(about_data.get('demistoVersion'))
+
+
+def arg_to_array(arg: Union[str, List[str]]) -> List[str]:
```
[the remark this answers is not in the document]
Comment (daryakoval) [reply demisto_sdk/commands/common/tools.py:1435]: done

File: demisto_sdk/commands/common/tools.py [hunk 9]
```diff
@@ -1430,3 +1430,24 @@ def get_demisto_version(demisto_client: demisto_client) -> str:
...
+        if isinstance(arg, list):
+            return arg
+        string_list = arg.split(',')
+        first = string_list[0]
+        last = string_list[-1]
+        if first[0] == '[':
+            string_list[0] = first[1:len(first)]
+            string_list[-1] = last[0:-1]
+        return string_list
+    return []
```
[the remark this answers is not in the document]
Comment (daryakoval) [reply demisto_sdk/commands/common/tools.py]: done

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,5 @@
 # Changelog
-* Added validation for approved content pack use-cases and tags.
+* Added validation for approved content pack usecases and tags.
```
Comment (ShahafBenYakir) [inline CHANGELOG.md:2]:
```suggestion
* Added validation for approved content pack use-cases and tags.
```

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**